src/components/ColumnLeft/CredentialSetItem/CredentialSetItem.js

```diff
@@ -15,11 +20,51 @@ export default function CredentialSetItem({
       type,
       title,
       subtitle,
-      expiration_dateint: d,
+      expiration_dateint: expiry,
     },
   },
   index,
+  records,
 }) {
+  const [isDisconnected, setIsDisconnected] = useState(false)
+  const [statusConnId, errorConnId, startFetchConnIdHandler] = useGetConn()
+  const [errorProposal, startFetchHandler] = usePostPresentProofSendProposal()
+  const [statusSendProposal, setStatusSendProposal] = useState('')
+
+  const toggleSuccessShowHandler = () => {
+    setStatusSendProposal('idle')
+  }
+
+  const onSubmit = (connectionId) => {
+    startFetchHandler(
+      origin,
+      connectionId,
+      id,
+      type,
+      cred_def_id,
+      setStatusSendProposal,
+      () => {}
+    )
+  }
+
+  const submitHandler = () => {
+    const connIdCb = (connectionId) => {
+      if (!connectionId) {
+        setIsDisconnected(true)
+        return
+      }
+      onSubmit(connectionId)
+    }
+    setIsDisconnected(false)
+    startFetchConnIdHandler(origin, connIdCb)
+  }
+
+  const alreadyProposed = () => {
+    return records?.some((record) => {
+      return record.pres_proposal?.comment === cred_def_id
```

Comment:
I've used `comment` on the proposal to easily link a proposal with an existing credential (the test pass from the issuer)

Reply:
Could this be a hook and happen in useEffect is just because that this looks component specific? or even 
```
...
const [ hook, updateHook] = React.usState({
  status: 'dead',
  isDisconnected: true,
  alreadyProposed: records?.some((record) => record.pres_proposal?.comment === cred_def_id),
});
```

Just an idea

Reply:
Going to move it into a new component